libs/chonky/src/embedding/hugging_face_api.rs

```diff
@@ -0,0 +1,177 @@
+use std::path::Path;
+
+use error_stack::{Report, ResultExt as _};
+use reqwest::{
+    Client,
+    header::{AUTHORIZATION, CONTENT_TYPE, HeaderMap, HeaderValue},
+};
+use serde::Deserialize;
+use tokio::fs;
+
+use crate::ChonkyError;
+
+#[derive(Deserialize, Debug)]
+pub struct BoundingBox {
+    pub xmin: f32,
+    pub ymin: f32,
+    pub xmax: f32,
+    pub ymax: f32,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct TablePrediction {
+    pub score: f32,
+    #[serde(rename = "box")]
+    pub bounding_box: BoundingBox,
+}
+
+//for now just have it as environment variable
+fn get_hugging_face_token() -> Result<String, Report<ChonkyError>> {
+    std::env::var("HUGGING_FACE_TOKEN").change_context(ChonkyError::HuggingFaceAPI)
+}
+
+async fn get_binary_image_data(
+    image_path: impl AsRef<Path> + core::marker::Send + core::marker::Sync,
+) -> Result<Vec<u8>, Report<ChonkyError>> {
+    fs::read(image_path)
+        .await
+        .change_context(ChonkyError::ImageError)
+}
+
+fn extract_bounding_boxes(json_payload: &str) -> Result<Vec<TablePrediction>, Report<ChonkyError>> {
+    serde_json::from_str(json_payload).change_context(ChonkyError::HuggingFaceAPI)
+}
+
+/// A function that calls `HuggingFace` Serverless Inference API to perform
+/// table recognition on a given image and returns the vectors of bounding boxes of these tables
+///
+/// # Errors
+///
+/// [`ChonkyError::HuggingFaceAPI`] when there are HTTP request errors
+pub async fn make_table_recognition_request(
+    image_path: impl AsRef<Path> + core::marker::Send + core::marker::Sync,
```

Comment:
`Send` and `Sync` are in the prelude.
```suggestion
    image_path: impl AsRef<Path> + Send + Sync,
```